test/integration/scheduler/priorities_test.go

```diff
@@ -234,3 +243,121 @@ func makeContainersWithImages(images []string) []v1.Container {
+			t.Fatalf("Adding labels to node failed: %v", err)
+		}
+	}
+
+	// Taint the 0th node
+	taint := v1.Taint{
+		Key:    "k1",
+		Value:  "v1",
+		Effect: v1.TaintEffectNoSchedule,
+	}
+	if err = addTaintToNode(cs, nodes[0].Name, taint); err != nil {
+		t.Fatalf("Adding taint to node failed: %v", err)
+	}
+	if err = waitForNodeTaints(cs, nodes[0], []v1.Taint{taint}); err != nil {
+		t.Fatalf("Taint not seen on node: %v", err)
+	}
+
+	pause := imageutils.GetPauseImageName()
+	tests := []struct {
+		name string
+		pod  *v1.Pod
+		pods []*v1.Pod
+		fits bool
+		want []string // nodes expected to schedule onto
+	}{
+		// note: naming starts at index 0
+		// the symbol ~X~ means that node is infeasible
+		{
+			name: "place pod on a ~0~/1/2/3 cluster with MaxSkew=1, node-1 is the preferred fit",
+			pod: st.MakePod().Namespace(ns).Name("p").Label("foo", "").Container(pause).
+				SpreadConstraint(1, "node", softSpread, st.MakeLabelSelector().Exists("foo").Obj()).
+				Obj(),
+			pods: []*v1.Pod{
+				st.MakePod().Namespace(ns).Name("p1").Node("node-1").Label("foo", "").Container(pause).Obj(),
+				st.MakePod().Namespace(ns).Name("p2a").Node("node-2").Label("foo", "").Container(pause).Obj(),
+				st.MakePod().Namespace(ns).Name("p2b").Node("node-2").Label("foo", "").Container(pause).Obj(),
+				st.MakePod().Namespace(ns).Name("p3a").Node("node-3").Label("foo", "").Container(pause).Obj(),
+				st.MakePod().Namespace(ns).Name("p3b").Node("node-3").Label("foo", "").Container(pause).Obj(),
+				st.MakePod().Namespace(ns).Name("p3c").Node("node-3").Label("foo", "").Container(pause).Obj(),
+			},
+			fits: true,
+			want: []string{"node-1"},
+		},
+		{
+			name: "combined with hardSpread constraint on a ~4~/0/1/2 cluster",
+			pod: st.MakePod().Namespace(ns).Name("p").Label("foo", "").Container(pause).
+				SpreadConstraint(1, "node", softSpread, st.MakeLabelSelector().Exists("foo").Obj()).
+				SpreadConstraint(1, "zone", hardSpread, st.MakeLabelSelector().Exists("foo").Obj()).
+				Obj(),
+			pods: []*v1.Pod{
+				st.MakePod().Namespace(ns).Name("p0a").Node("node-0").Label("foo", "").Container(pause).Obj(),
+				st.MakePod().Namespace(ns).Name("p0b").Node("node-0").Label("foo", "").Container(pause).Obj(),
+				st.MakePod().Namespace(ns).Name("p0c").Node("node-0").Label("foo", "").Container(pause).Obj(),
+				st.MakePod().Namespace(ns).Name("p0d").Node("node-0").Label("foo", "").Container(pause).Obj(),
+				st.MakePod().Namespace(ns).Name("p2").Node("node-2").Label("foo", "").Container(pause).Obj(),
+				st.MakePod().Namespace(ns).Name("p3a").Node("node-3").Label("foo", "").Container(pause).Obj(),
+				st.MakePod().Namespace(ns).Name("p3b").Node("node-3").Label("foo", "").Container(pause).Obj(),
+			},
+			fits: true,
+			want: []string{"node-2"},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			allPods := append(tt.pods, tt.pod)
+			defer cleanupPods(cs, t, allPods)
+			for _, pod := range tt.pods {
```

Comment:
The pod creation can go to a helper function that can be reused in predicates_test